controllers/vrg_volrep.go

```diff
@@ -210,6 +210,17 @@ func (v *VRGInstance) preparePVCForVRProtection(pvc *corev1.PersistentVolumeClai
 		return !requeue, !skip
 	}
 
+	// if PVC is protected by another VRG, skip this PVC
+	// there might be PVCs that don't have a conflicting owner
+	if pvcVRGOwner, found := pvc.Annotations[pvcVRAnnotationOwnerKey]; found {
```

Comment:
At this point we have conflicting VRGs and should not allow progression of the reconcile, as PVCs maybe protected by either VRGs as they race in reconciliation.

Reply:
> I suggest we use locks, mapping them to a map with key based on --. This will ensure only one VRG is reconciled at a time, and serialize the conflict checks in the code (PVC ownership).

Could you elaborate a bit more on "mapping them to a map with key based on --". 

IIUC, this is a lock within the controller. Wouldn't that have an impact the processing time as there could be many VRGs in the system?

Reply:
The map of locks is [this](https://github.com/ceph/ceph-csi/blob/ec242d4cc8fd5464117c78567d95ed5bb1e6e8d0/internal/util/idlocker.go#L40-L65) pattern. This will serialize only those reconciles that conflict on the string that is common, and hence other VRG resources would reconcile in parallel.

Looks like I did not add a potential key example, it would be say `<namespace>-<label-selector-hash>` so that 2 VRGs in the same namespace having the same label selector will not reconcile in parallel.